src/components/assets/XcmCurrency.vue

```diff
@@ -106,33 +106,34 @@ export default defineComponent({
   },
   setup(props) {
     const t = computed(() => props.token);
-    const { tokenImage, tokenDetails, isXcmCompatible } = useXcmTokenDetails(t);
     const store = useStore();
 
+    const currentNetworkIdx = computed(() => {
+      const chainInfo = store.getters['general/chainInfo'];
+      const chain = chainInfo ? chainInfo.chain : '';
+      return getProviderIndex(chain);
+    });
+
     const isDisplayToken = computed<boolean>(() => {
-      // Todo: fetch the balance in relaychain
-      const isDisplay = Number(props.token.userBalance) > 0 || tokenDetails.value?.isXcmCompatible;
+      const token = getXcmToken({
+        id: t.value.id,
+        currentNetworkIdx: currentNetworkIdx.value,
+      });
+      const isDisplay = Number(props.token.userBalance) > 0 || !!token;
       return isDisplay || false;
     });
 
     const explorerLink = computed(() => {
       const chainInfo = store.getters['general/chainInfo'];
       const chain = chainInfo ? chainInfo.chain : '';
       const currentNetworkIdx = getProviderIndex(chain);
-
-      const astarBalanceUrl =
-        'https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.astar.network#/assets/balances';
-      const shidenBalanceUrl =
-        'https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.shiden.astar.network#/assets/balances';
-
+      const astarBalanceUrl = 'https://astar.subscan.io/assets/' + t.value.id;
```

Comment:
I recommend we move this to the Token API instead of directly reading from subscan.
What do you think @bobo-k2 ?

Reply:
@hoonsubin are you suggesting `token-API` to provide the URL link like  'https://astar.subscan.io/assets/' ?

Reply:
To be specific, have the token API also expose the clean version of registered assets (with the XC20 address and other information).
This is not critical, but I want to avoid having the Portal reference multiple remote sources.